### PR TITLE
promote validation

### PR DESCRIPTION
**Pull request by bekzod**
regarding #457

### CODE REVIEW

File: lib/rsvp/utils.js [hunk 1]
```diff
@@ -6,6 +6,10 @@ export function isFunction(x) {
   return typeof x === 'function';
 }
 
+export function isObject(x) {
+  return x !== undefined && x !== null && typeof x === 'object';
```
Comment (stefanpenner) [inline lib/rsvp/utils.js]: Undefined check is not needed as typeof object already includes that

Comment (bekzod) [inline lib/rsvp/utils.js]: my assumption was that checking for `undefined` is cheaper and `typeof` is expensive, so I am checking for undefined first (and if it is undefined it short circuits) am I wrong ?

Comment (bekzod) [inline lib/rsvp/utils.js]: reverted

**Comment by stefanpenner** [comment]
needs rebase, but this looks good.

**Pull request merged**